components/match2/commons/match_summary_base.lua

```diff
@@ -313,66 +365,131 @@ function MatchSummary:header(header)
-		:css('font-weight', 'bold')
-		:wikitext('Reset match')
+	return self.root
 end
 
-function MatchSummary:create()
-	self.root
-		:node(self.headerElement)
-		:node(Break():create())
-		:node(self.bodyElement)
-		:node(Break():create())
+---Default header function
+---@param match table
+---@param options {noScore: boolean?}?
+---@return MatchSummaryHeader
+function MatchSummary.createDefaultHeader(match, options)
+	options = options or {}
 
-	if self.resetBodyElement then
-		self.root
-			:node(self.resetHeaderNode or MatchSummary._fallbackResetHeader())
-			:node(Break():create())
-			:node(self.resetBodyElement)
-			:node(Break():create())
+	local header = MatchSummary.Header()
+
+	if options.noScore then
+		return header
+			:leftOpponent(header:createOpponent(match.opponents[1], 'left'))
+			:rightOpponent(header:createOpponent(match.opponents[2], 'right'))
 	end
 
-	if self.commentElement ~= nil then
-		self.root
-			:node(self.commentElement)
-			:node(Break():create())
+	return header
+		:leftOpponent(header:createOpponent(match.opponents[1], 'left'))
+		:leftScore(header:createScore(match.opponents[1]))
+		:rightScore(header:createScore(match.opponents[2]))
+		:rightOpponent(header:createOpponent(match.opponents[2], 'right'))
+end
+
+---Creates a match footer with vods if vods are set
+---@param match table
+---@return MatchSummaryFooter?
+function MatchSummary.createVodFooter(match)
+	local vods = {}
+	for index, game in ipairs(match.games) do
+		if game.vod then
+			vods[index] = game.vod
+		end
 	end
 
-	if self.footerElement ~= nil then
-		self.root:node(self.footerElement)
+	if Table.isEmpty(vods) and not match.vod then
+		return
 	end
 
-	return self.root
+	local footer = MatchSummary.Footer()
+	if match.vod then
+		footer:addElement(VodLink.display{
+			vod = match.vod,
+		})
+	end
+	for index, vod in pairs(vods) do
+		footer:addElement(VodLink.display{
+			gamenum = index,
+			vod = vod,
+		})
+	end
+
+	return footer
+end
+
+---Default createMatch function for usage in Custom MatchSummary
+---@param matchData table
+---@param CustomMatchSummary table
+---@return MatchSummaryMatch?
+function MatchSummary.createMatch(matchData, CustomMatchSummary)
+	if not matchData then
+		return
+	end
+
+	local match = Match()
+
+	local createHeader = CustomMatchSummary.createHeader or MatchSummary.createDefaultHeader
+	match:header(createHeader(matchData))
+
+	match:body(CustomMatchSummary.createBody(matchData))
+
+	if matchData.comment then
+		local comment = MatchSummary.Comment():content(matchData.comment)
+		match:comment(comment)
+	end
+	local createFooter = CustomMatchSummary.createFooter or MatchSummary.createVodFooter
+	local footer = createFooter(matchData)
+	if footer then
+		match:footer(footer)
+	end
+
+	return match
+end
+
+---Default getByMatchId function for usage in Custom MatchSummary
+---@param CustomMatchSummary table
+---@param args table
+---@return Html
+function MatchSummary.defaultGetByMatchId(CustomMatchSummary, args)
```

Comment:
idea is to have a basic function that should work for most wikis already
the `createBody` function is usually very wiki specific and hence remains on the Customs imo